### ISSUE TITLE
Failed to load list of diff files

### ISSUE THREAD

**Issue by jpietrangelo** [zaggino/brackets-git#987]
1. Clicked init
2. Added one file to .gitignore
3. Clicked on 'Show History' icon.
4. Fatal error popped up.

--------------------------------

*Brackets 1.2.0-15697 (mac), Git 2.2.1 & Brackets-Git 0.14.19*

**Failed to load list of diff files**

```
Error: fatal: ambiguous argument '978f3f22e1d74d244123cd739c8e977c8eb6c454^..978f3f22e1d74d244123cd739c8e977c8eb6c454': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

```
call: git diff --no-ext-diff --name-only 978f3f22e1d74d244123cd739c8e977c8eb6c454^..978f3f22e1d74d244123cd739c8e977c8eb6c454
Error: fatal: ambiguous argument '978f3f22e1d74d244123cd739c8e977c8eb6c454^..978f3f22e1d74d244123cd739c8e977c8eb6c454': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
    at Object.exports.toError (file:///Users/jmp/Library/Application%20Support/Brackets/extensions/user/zaggino.brackets-git/src/ErrorHandler.js:162:19)
    at NodeConnection.<anonymous> (file:///Users/jmp/Library/Application%20Support/Brackets/extensions/user/zaggino.brackets-git/src/Cli.js:176:44)
    at j (file:///Applications/Brackets.app/Contents/www/thirdparty/thirdparty.min.js:559:26911)
    at Object.k.fireWith (file:///Applications/Brackets.app/Contents/www/thirdparty/thirdparty.min.js:559:27724)
    at NodeConnection._receive (/utils/NodeConnection.js:492:34)
```

**Comment by jpietrangelo** [comment]
Sorry. Made a mistake above. The fatal error did not pop up when I clicked the 'Show History' icon. It pops up if I click on the actual 'Initial commit' line in the window.

**Comment by zaggino** [comment]
That's an extension bug, `from` and `to` can't be the same, will look into fixing it for next version. Once you do more commits, it should be ok.

**Comment by MarcelGerber** [comment]
Sorry, that's my fault caused by https://github.com/zaggino/brackets-git/pull/975/files.
@zaggino From and to aren't the same -- from is `commit^` (notice the `^`) and to is `commit`. It's just that, as `commit` is the first commit in the repo/branch, there's no such thing as `commit^`

**Comment by jpietrangelo** [comment]
Makes sense. Thanks for the quick reply.

Jim

**Comment by jpietrangelo** [comment]
Makes perfect sense. Thanks Martin.

Jim

**Comment by MarcelGerber** [comment]
@zaggino I can't say for sure as I've only done little testing, but `git diff-tree -p -m hash` seems to be the right thing to do then.

**Comment by zaggino** [comment]
Thanks @MarcelGerber , I'll look into this

**Comment by zaggino** [comment]
Fixed in version `0.14.21`, not able to update extension registry right now, but available through https://github.com/zaggino/brackets-npm-registry

**Comment by MarcelGerber** [comment]
@zaggino Your fix is definitely not the correct one -- that's what I got for Brackets' initial commit (adobe/brackets@637d7f):
![image](https://cloud.githubusercontent.com/assets/2641501/7022652/4c33c6ea-dd2e-11e4-9812-c5745948e98a.png)

Also, you should handle `git difftool` the same way you do with `git diff`

**Comment by zaggino** [comment]
Strange, it worked for one of my repos that tested it with. Will look into it again and make sure I test it on Brackets repo.

**Comment by zaggino** [comment]
Fixed in #989
